.github/workflows/terragrunt-plan-staging.yml

```diff
@@ -141,6 +144,16 @@ jobs:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
           terragrunt: "true"
 
+      - name: Terragrunt plan sns
+        if: ${{ steps.filter.outputs.sns == 'true' || steps.filter.outputs.common == 'true' }}
+        uses: cds-snc/terraform-plan@v2
+        with:
+          directory: "env/production/sns"
```

Comment:
`env/staging/sns` - if you're curious, this is why you were getting the plan failure since it was trying to use the production environment config, which expected a `TARGET_VERSION` env variable.